CONTRIBUTING.md

```diff
@@ -356,3 +357,77 @@ When integrating 3rd-party code into the code base, use the following scaffold:
 
 (code here)
 ```
+
+Unit Tests
+----------
+
+- Every new feature must come with unit tests.
+
+- The filename and path should mirror the component under test. For example,
+  the component `vast/detail/feature.hpp` should have a test file called
+  `test/detail/feature.cpp`.
+
+- The include order in unit tests resembles the order for standard headers,
+  except that unit test includes and the suite definition comes at the top.
+
+- Make judicious use of *fixtures* for prepping your test environment.
+
+- The snippet below illustrates a simple example for a new component
+  `vast/foo.hpp` that would go into `test/foo.cpp`.
+
+  ```cpp
+  /******************************************************************************
+   *                    _   _____   __________                                  *
+   *                   | | / / _ | / __/_  __/     Visibility                   *
+   *                   | |/ / __ |_\ \  / /          Across                     *
+   *                   |___/_/ |_/___/ /_/       Space and Time                 *
+   *                                                                            *
+   * This file is part of VAST. It is subject to the license terms in the       *
+   * LICENSE file found in the top-level directory of this distribution and at  *
+   * http://vast.io/license. No part of VAST, including this file, may be       *
+   * copied, modified, propagated, or distributed except according to the terms *
+   * contained in the LICENSE file.                                             *
+   ******************************************************************************/
+
+  #define SUITE foo
+  #include "test.hpp"
+  #include "fixtures/my_fixture.hpp"
+
+  #include <iostream>     // standard library includes
+
+  #include <caf/...>      // CAF includes
+
+  #include "vast/foo.hpp" // VAST includes
```

Comment:
This is the unit-under-test. Why did you opt against including this first?

Reply:
I think this was an oversight. I wanted to write `#include "vast/..."` but my mental context was still within `foo`. Will fix.

Reply:
I think this include should come first in source fies.